app/models/host/managed.rb

```diff
@@ -926,6 +927,17 @@ def apply_compute_profile(modification)
 
   private
 
+  def owner_taxonomies_match
+    if self.owner.is_a?(User)
+      if Organization.organizations_enabled && self.organization_id && !self.owner.my_organizations.where(:id => self.organization_id).exists?
```

Comment:
+1 for keeping the public methods/scopes outside of meta-programming. It makes it much easier to navigate through the code later.

I see the problem is we have `my_organizations` and `taxonomy_and_child_ids` that seem to do similar things. We should settle on using one of them publicly and stick on it. Since the `my_organizations`/`my_locations` takes into account the `admin` flag and it's more powerful, I would prefer using them as public methods, and hiding the `taxonomy_and_child_ids` into private method of `Taxonomy` class. This way it would be clear in the future which method/scope to use.